src/test/java/edu/cornell/kfs/sys/web/mock/MockMvcWebServerExtension.java

```diff
@@ -52,31 +57,59 @@
  * 
  * This class is meant to be used as a *programmatically-configured* JUnit 5 extension.
  * Your test class should set up an instance of this extension by declaring an appropriate
- * non-private instance field annotated with "@RegisterExtension".
+ * non-private instance field or static field annotated with "@RegisterExtension".
  * 
  * Prior to each test method being called, this class's internal MockMvc instance needs
- * to be initialized by the caller. Such setup can happen within an annotated "@BeforeEach"
- * method on the test class. The simplest way to perform the initialization is by calling
+ * to be initialized by the caller. Such setup can happen within a test class's annotated
+ * "@BeforeEach" method (for an instance-level extension) or its "@BeforeAll" method
+ * (for a class-level extension). The simplest way to perform the initialization is by calling
  * the initializeStandaloneMockMvcWithControllers() method and passing in the MVC Controllers
  * that you want to use. Alternatively, you can use the setMockMvc() method if you need
  * finer control over the MockMvc instance's configuration.
  * 
  * To retrieve the HTTP server's URL, call the getServerUrl() method from within your test class's
- * "@BeforeEach"-annotated method(s). The returned URL will *not* contain a trailing slash.
+ * "@BeforeEach"-annotated or "@BeforeAll"-annotated method(s). The returned URL will *not* contain
+ * a trailing slash.
+ * 
+ * If the extension is set up at the class level, the test server will be initialized prior to
+ * running the class's test suite and will be shut down after running the suite. If the configured
+ * controller instances need to reset their state in between test methods, they can implement
+ * the ResettableController interface and then this extension will reset them after each run
+ * of a test method. (Note that this auto-reset feature is ignored if the extension was set up
+ * as an instance variable or if the MVC configuration was set up via the setMockMvc() method.)
+ * 
+ * Note that if this extension is set up at the class/static level, it is strongly recommended
+ * that the test class be configured for SAME_THREAD execution mode so that the test methods
+ * don't interfere with each other.
  */
-public class MockMvcWebServerExtension implements BeforeEachCallback, BeforeTestExecutionCallback, AfterEachCallback {
+public class MockMvcWebServerExtension implements BeforeEachCallback, BeforeTestExecutionCallback, AfterEachCallback,
```

Comment:
This is interesting, maybe you could demo how to use this class in the Developer's Bookclub?

Reply:
Good idea, I've added a note to the main Bookclub Confluence page to discuss this further at one of our future Bookclub meetings.